File: lib/data/isar/models/schedule_event.dart
import 'package:isar/isar.dart';

part 'schedule_event.g.dart';

enum ScheduleEventType {
  practice,
  game,
}

@collection
class ScheduleEvent {
  Id id = Isar.autoIncrement;

  @Index(unique: true)
  late String uuid;

  late String teamId;

  @enumerated
  late ScheduleEventType type;

  late DateTime startsAt;

  DateTime? endsAt;

  String? location;

  /// Opponent name; typically used when type == game.
  String? opponent;

  String? notes;

  late DateTime createdAt;

  late DateTime updatedAt;

  /// User ID of last updater (cloud-ready).
  String? updatedByUserId;

  /// Soft delete for sync (cloud-ready).
  DateTime? deletedAt;

  /// Schema version for migrations (cloud-ready).
  late int schemaVersion;

  ScheduleEvent();

  ScheduleEvent.create({
    required this.uuid,
    required this.teamId,
    required this.type,
    required this.startsAt,
    this.endsAt,
    this.location,
    this.opponent,
    this.notes,
    DateTime? createdAt,
    DateTime? updatedAt,
    this.updatedByUserId,
    this.deletedAt,
    this.schemaVersion = 1,
  })  : createdAt = createdAt ?? DateTime.now(),
        updatedAt = updatedAt ?? DateTime.now();
}

File: backend/src/models/ScheduleEvent.js
const mongoose = require('mongoose');
const { v4: uuidv4 } = require('uuid');

const scheduleEventSchema = new mongoose.Schema(
  {
    uuid: { type: String, required: true, unique: true, default: () => uuidv4() },
    teamId: { type: String, required: true },
    type: { type: String, required: true, enum: ['practice', 'game'] },
    startsAt: { type: Date, required: true },
    endsAt: { type: Date, default: null },
    location: { type: String, default: null },
    opponent: { type: String, default: null },
    notes: { type: String, default: null },
    updatedAt: { type: Date, default: () => new Date() },
    updatedBy: { type: String, default: null },
    deletedAt: { type: Date, default: null },
    schemaVersion: { type: Number, default: 1 },
  },
  { timestamps: false }
);

scheduleEventSchema.index({ teamId: 1, updatedAt: 1 });
scheduleEventSchema.index({ teamId: 1, deletedAt: 1 });

const ScheduleEvent = mongoose.model('ScheduleEvent', scheduleEventSchema);
module.exports = { ScheduleEvent };
